test/unit/vl-pill.test.html

```diff
@@ -14,84 +14,139 @@
+    test('closable pill stuurt event uit wanneer er op de close knop gedrukt wordt', (done) => {
+      const pill = fixture('vl-pill-fixture');
 
-      // valid values
-      ["success","warning","error"].forEach((val) => {
-        assert.isFalse(checkablePill.classList.contains("vl-pill--"+val));
-        checkablePill.setAttribute('type',val);
-        assert.isTrue(checkablePill.classList.contains("vl-pill--"+val));
-        checkablePill.removeAttribute('type');
-        assert.isFalse(checkablePill.classList.contains("vl-pill--"+val));
+      pill.setAttribute('closable', '');
+      pill.addEventListener(VlPill.EVENTS.close, (event) => {
+        event.composed.should.be.true;
+        event.bubbles.should.be.true;
+        done();
       });
 
-      // invalid value
-      let val = "blabla";
-      checkablePill.setAttribute('type',val);
-      assert.isFalse(checkablePill.classList.contains("vl-pill--"+val));
+      pill._closeButton.click();
     });
 
-    test('checkable bevat input element van het type checkbox met juiste classes', () => {
-      const checkablePill = fixture('vl-label-pill-fixture');
-      let inputElement = checkablePill._inputElement;
-      should.exist(inputElement);
-      assert.isTrue(inputElement.getAttribute('type') == 'checkbox')
-      assert.isTrue(inputElement.classList.contains('vl-pill--checkable__checkbox'))
+    test('checkable pill bevat de juiste class en een input element van het type checkbox met de juiste class', () => {
+      const pill = fixture('vl-pill-fixture');
+      const shadowRoot = pill.shadowRoot;
+
+      should.not.exist(shadowRoot.querySelector(".vl-pill--checkable"));
+
+      pill.setAttribute('checkable', '');
+      const checkbox = pill._checkbox;
+      assert.isTrue(checkbox.getAttribute('type') === 'checkbox');
+      assert.isTrue(checkbox.classList.contains('vl-pill--checkable__checkbox'));
+      should.exist(shadowRoot.querySelector(".vl-pill--checkable"));
+
+      pill.removeAttribute('checkable');
+      should.not.exist(shadowRoot.querySelector("input"));
+      should.not.exist(shadowRoot.querySelector(".vl-pill--checkable"));
     });
 
-    test('checkable bevat een span element', () => {
-      const checkablePill = fixture('vl-label-pill-fixture');
-      should.exist(checkablePill.querySelector('span'));
+    test('checkable pill bevat een span element', () => {
+      const pill = fixture('vl-pill-fixture');
+      const shadowRoot = pill.shadowRoot;
+
+      pill.setAttribute('checkable', '');
+      should.exist(shadowRoot.querySelector('span'));
     });
+
+    test('checkable pill kan via "checked" uit- en aangevinkt worden', () => {
+      const pill = fixture('vl-pill-fixture');
+
+      pill.setAttribute('checkable', '');
+      pill.checked.should.be.false;
+
+      pill.checked = true;
+      pill.checked.should.be.true;
+
+      pill.checked = false;
+      pill.checked.should.be.false;
```

Comment:
is da nie wa dwaas?

Reply:
Nee, dat is niet dwaas. setter != getter, daarnaast wordt de getter nog eens getest bij het clicken op de pill dus we weten dat het het juiste gaat teruggeven.
@coemans praat @pascalriquier eens wat wijsheid bij aub.